### DISCUSSION TITLE
How can I create a Planner Task Checklist?

### DISCUSSION THREAD

**Discussion by Gopher2000** [microsoftgraph/msgraph-sdk-dotnet#2083] [Q&A] [answered]
The example on creating a planner task checklist on [](https://learn.microsoft.com/en-us/graph/api/plannertaskdetails-update?view=graph-rest-1.0&tabs=csharp) throws a Microsoft.Graph.Models.ODataErrors.ODataError exception .   

[PlannerTests.cs](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/tests/Microsoft.Graph.DotnetCore.Test/Requests/Functional/PlannerTests.cs) hasn't been updated for the 5.X version yet.   

I'm using the following code: 

`        public async static Task<PlannerTaskDetails> UpdateDetailsOfTaskAddChecklist(GraphServiceClient graphClient, PlannerTask plannerTask, string[] checklistItems)
        {
            PlannerTaskDetails requestBody = await graphClient.Planner.Tasks[plannerTask.Id].Details.GetAsync();
            string eTag = requestBody.AdditionalData["@odata.etag"].ToString();

            requestBody.PreviewType = PlannerPreviewType.Checklist;

            if (requestBody.Checklist == null)
            {
                requestBody.Checklist = new PlannerChecklistItems();
                requestBody.Checklist.AdditionalData = new Dictionary<string, object>();
            }

            foreach (var i in checklistItems)
            {
                requestBody.Checklist.AdditionalData.Add(Guid.NewGuid().ToString(),
                    new
                    {
                        OdataType = "microsoft.graph.plannerChecklistItem",
                        Title = i,
                        IsChecked = false
                    }
                 );
            }

            var result = await graphClient.Planner.Tasks[plannerTask.Id].Details.PatchAsync(requestBody, (requestConfiguration) =>
            {
                requestConfiguration.Headers.Add("Prefer", "return=representation");
                requestConfiguration.Headers.Add("If-Match", eTag);
            });

            return result;
        }`

**Comment by andrueastman** [comment, marked as answer]
This should be resolved in the latest version 5.24.0 by specifying the type `PlannerChecklistItem`

```cs
        foreach (var i in checklistItems)
        {
            requestBody.Checklist.AdditionalData.Add(Guid.NewGuid().ToString(),
                new PlannerChecklistItem
                {
                    OdataType = "microsoft.graph.plannerChecklistItem",
                    Title = i,
                    IsChecked = false
                }
             );
        }
```

**Reply by Gopher2000** [reply]
Thanks, it works now. 
`foreach (var i in checklistItems)
            {
                requestBody.Checklist.AdditionalData.Add(Guid.NewGuid().ToString(),
                    new PlannerCheckListItem
                    {
                        OdataType = "microsoft.graph.plannerChecklistItem",
                        Title = i,
                        IsChecked = false
                    }
                 );
            }`
